src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketHgSCMBuilder.java

```diff
@@ -140,6 +150,43 @@ public BitbucketSCMSource scmSource() {
         return Collections.unmodifiableList(cloneLinks);
     }
 
+    /**
+     * Configures the {@link IdCredentials#getId()} of the {@link Credentials} to use when connecting to the
+     * {@link #source()}
+     *
+     * @param credentialsId the {@link IdCredentials#getId()} of the {@link Credentials} to use when connecting to
+     *                      the {@link #source()} or {@code null} to let the git client choose between providing its own
+     *                      credentials or connecting anonymously.
+     * @param protocol the {@link BitbucketRepositoryProtocol} of the {@link Credentials} to use or {@code null}
+     *                 to detect the the protocol based on the credentialsId. Defaults to HTTP if credentials are
+     *                 {@code null}.  Enables support for blank SSH credentials.
+     * @return {@code this} for method chaining.
+     */
+    @NonNull
+    public BitbucketHgSCMBuilder withCredentialsId(String credentialsId, BitbucketRepositoryProtocol protocol) {
+        if (StringUtils.isNotBlank(credentialsId)) {
+            StandardCredentials credentials = BitbucketCredentials.lookupCredentials(
+                scmSource.getServerUrl(),
+                scmSource.getOwner(),
+                credentialsId,
+                StandardCredentials.class
+            );
+
+            if (protocol == null) {
+                protocol = credentials instanceof SSHUserPrivateKey
+                    ? BitbucketRepositoryProtocol.SSH
+                    : BitbucketRepositoryProtocol.HTTP;
+            }
+        } else if (protocol == null) {
+            // If we set credentials to empty reset the type to HTTP.
+            // To set the build to use empty SSH credentials, call withProtocol after setting credentials
+            protocol = BitbucketRepositoryProtocol.HTTP;
+        }
+
+        this.protocol = protocol;
+        return withCredentialsId(credentialsId);
```

Comment:
It feels very unnecessary to do two credential lookups, since depending on implementation of the credentials store could be network requests. But I guess this is implemented in the mercurial plugin so not easy to get changed? Maybe it's not even doing a lookup?

Reply:
Lookup? This is setting the value not looking it up.

Reply:
Just to be clear @rsandell this is the piece of code from the extended class :smile: 
https://github.com/jenkinsci/git-plugin/blob/6d22cc0188629f941a5751c4b0d628ff91557dc8/src/main/java/jenkins/plugins/git/GitSCMBuilder.java#L264-L267